app/vmalert/rule/group_test.go

```diff
@@ -353,6 +353,8 @@ func TestGroupStart(t *testing.T) {
 		}
 	}
 
+	// wait the group to be started
+	time.Sleep(100 * time.Millisecond)
```

Comment:
nit: Is it possible to synchronize this in order to avoid adding time.Sleep?

Reply:
I don't see a good way, the below `waitForIterations()` call must wait until `g.metrics.iterationTotal` is created.
But if I add nil check for `g.metrics.iterationTotal` like
```
--- a/app/vmalert/rule/group_test.go
+++ b/app/vmalert/rule/group_test.go
@@ -354,7 +354,9 @@ func TestGroupStart(t *testing.T) {
        }
 
        // wait the group to be started
-       time.Sleep(100 * time.Millisecond)
+       for g.metrics.iterationTotal == nil {
+               time.Sleep(10 * time.Millisecond)
+       }
        // wait for multiple evaluation iterations
        waitForIterations(4, evalInterval)
```
I get data race for `g.metrics.iterationTotal`. 
Do you have a better idea?

Reply:
Would it be possible to use global metrics set to query for this specific metric?

Reply:
I don't think so, the metricsSet only has functions `GetOrCreatexxx(name string)`, no `Getxxx(name string)`.